src/mapping_navigation/scripts/pure_pursuit.py

```diff
@@ -0,0 +1,152 @@
+"""
+Code adapted from
+
+Path tracking simulation with pure pursuit steering and PID speed control.
+author: Atsushi Sakai (@Atsushi_twi)
+        Guillaume Jacquenot (@Gjacquenot)
+"""
+import numpy as np
+import math
+from typing import List, Tuple
+from map_information import *
+
+class State:
+
+    """
+    Contains the car information needed in pure pursuit
+    The cartesian coordinates for location
+    The yaw of the car
+    The cars wheel base
+    """
+    def __init__(self, wheel_base: float, x: float=0.0, y:float=0.0, yaw:float=0.0, velocity:float=0.0):
+        self.wheel_base = wheel_base
```

Comment:
Was wondering what wheel base is for, and later in the file rear_x is for?

Reply:
pure pursuit does its calculations from the back of the car, so wheel base is used to offset where the coordinates are, now that I think of it though, I think the gps sensor takes from the center, so is should be divided by 2